From 455834befebac359493dbe0d843cab01e44031cf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Domen=20Ko=C5=BEar?= <domen@dev.si>
Date: Wed, 21 Dec 2022 15:39:38 +0000
Subject: [PATCH] DEVENV: move the error message to the bottom when showing
 traces

---
 src/libutil/error.cc | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/libutil/error.cc b/src/libutil/error.cc
index e01f064484c..b4064a5297a 100644
--- a/src/libutil/error.cc
+++ b/src/libutil/error.cc
@@ -419,15 +419,18 @@ std::ostream & showErrorInfo(std::ostream & out, const ErrorInfo & einfo, bool s
         oss << "\n" << prefix;
     }
 
-    oss << einfo.msg << "\n";
 
     printPosMaybe(oss, "", einfo.pos);
 
+    oss << "\n" << prefix << einfo.msg << "\n";
+
     auto suggestions = einfo.suggestions.trim();
     if (!suggestions.suggestions.empty()) {
+        oss << "\n";
         oss << "Did you mean " <<
             suggestions.trim() <<
             "?" << std::endl;
+        oss << "\n";
     }
 
     out << indent(prefix, std::string(filterANSIEscapes(prefix, true).size(), ' '), chomp(oss.str()));
