From 67da888fc977ba29d00ecf71ca69abb0de22ac93 Mon Sep 17 00:00:00 2001
From: Sander <hey@sandydoo.me>
Date: Thu, 12 Jun 2025 00:48:00 +0200
Subject: [PATCH] devenv: remove inactionable inefficient double copy warning

---
 src/libexpr/paths.cc | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/src/libexpr/paths.cc b/src/libexpr/paths.cc
index fa8ebea48..2bab3361c 100644
--- a/src/libexpr/paths.cc
+++ b/src/libexpr/paths.cc
@@ -60,11 +60,6 @@ std::string EvalState::computeBaseName(const SourcePath & path)
 {
     if (path.accessor == rootFS) {
         if (auto storePath = store->maybeParseStorePath(path.path.abs())) {
-            warn(
-                "Performing inefficient double copy of path '%s' to the store. "
-                "This can typically be avoided by rewriting an attribute like `src = ./.` "
-                "to `src = builtins.path { path = ./.; name = \"source\"; }`.",
-                path);
             return std::string(
                 fetchToStore(fetchSettings, *store, path, FetchMode::DryRun, storePath->name()).to_string());
         }
-- 
2.49.0

